tests/unit/extensions/test_gunicorn.py

```diff
@@ -309,6 +310,38 @@ def test_flask_extension_no_requirements_txt_error(tmp_path):
     with pytest.raises(ExtensionError) as exc:
         extensions.apply_extensions(tmp_path, flask_input_yaml)
     assert "requirements.txt" in str(exc)
+    assert "app.py" not in str(exc)
+
+
+@pytest.mark.usefixtures("flask_extension")
+def test_flask_extension_requirements_txt_no_flask_error(tmp_path):
+    (tmp_path / "app.py").write_text("app = object()")
+    (tmp_path / "requirements.txt").write_text("")
+    flask_input_yaml = {
+        "extensions": ["flask-framework"],
+        "base": "bare",
+        "build-base": "ubuntu@22.04",
+        "platforms": {"amd64": {}},
+    }
+    with pytest.raises(ExtensionError) as exc:
+        extensions.apply_extensions(tmp_path, flask_input_yaml)
+    assert "requirements.txt" in str(exc)
+    assert "flask package" in str(exc)
+    assert "app.py" not in str(exc)
+
+
+@pytest.mark.usefixtures("flask_extension")
+def test_flask_extension_no_requirements_txt_no_app_py_error(tmp_path):
+    flask_input_yaml = {
+        "extensions": ["flask-framework"],
+        "base": "bare",
+        "build-base": "ubuntu@22.04",
+        "platforms": {"amd64": {}},
+    }
+    with pytest.raises(ExtensionError) as exc:
+        extensions.apply_extensions(tmp_path, flask_input_yaml)
+    assert "requirements.txt" in str(exc)
```

Comment:
I know that this is picky but can you test against the *exact* string of the error message? Just so we're sure it has nice formatting.

Reply:
Sure. I did not want to copy-paste the message, but it makes sense. Update to match against the exact message.